### ISSUE TITLE
[ENHANCEMENT] WinReg file associations

### ISSUE THREAD

**Issue by NullPlane** [SpartanJ/ecode#192]
An installer would be nice but I do appreciate the fact that ecode is a portable app too.  
Maybe some script to give the folder it is located in and then set an array of associations  
containing all possible associations that the editor may open?

**Comment by SpartanJ** [comment]
Yes, I was thinking in implementing this in the installer but it might be better to implement it at the application level.

**Comment by NullPlane** [comment]
Once there is an installer an actual checkbox would be convenient but to be sure app level is definitely cooler for those who use it portably.

**Comment by ghost** [comment]
> An installer would be nice but I do appreciate the fact that ecode is a portable app too. Maybe some script to give the folder it is located in and then set an array of associations containing all possible associations that the editor may open?

Are you thinking about the similar settings in 7-Zip? They already know all the archive types 7-Zip can open beforehand. Ecode, on the other hand, can handle any text files.

**Comment by NullPlane** [comment]
> > An installer would be nice but I do appreciate the fact that ecode is a portable app too. Maybe some script to give the folder it is located in and then set an array of associations containing all possible associations that the editor may open?
> 
> Are you thinking about the similar settings in 7-Zip? They already know all the archive types 7-Zip can open beforehand. Ecode, on the other hand, can handle any text files.

If there is an installer for Windows, it usually reigsters apps for default options etc.. ecode is also portable and does not have an installer for Windows yet. Yet, ecode is also an app that craves proper portability. For a short-term solution, I imagined a _pwsh_ or _cmd_ script that sets associations so ecode at least pops up in first-time openwiths. Currently, you'd have to manually search for ecode's installation folder on every new file to set it as default or use [PORTABLE Registrator](https://github.com/SiL3NC3/PortableRegistrator "PORTABLE Registrator on GitHub") to have the system ask for it. That could be automated with a predelivered script that takes ecode's location and sets this up

***Long-term, however,*** as ecode is so nicely ambiguous about "where will it run" and can be out-of-the-box compiled and -used for some obscure OSes like Haiku or even Android, it would be great to have a solution in-editor to set these associations, which adjusts actions depending on the OSes. Here, I am looking for something like how [ImageGlass](https://github.com/d2phap/ImageGlass "ImageGlass on GitHub") or Media Player Classic ([MPC-BE](https://github.com/Aleksoid1978/MPC-BE "MPC-BE on GitHub") or [MPC-HC](https://github.com/clsid2/mpc-hc "mpc-hc on GitHub")) do it through a workflow and UI that is delivered with the app. (Screenshots below)

Of course, installers can automate that on any system or package manager however, the fine-grained control being delivered with the app is something I could imagine for ecode.

<img width="1279" height="788" alt="Image" src="https://github.com/user-attachments/assets/ee6814e9-6033-4972-8f33-cf5b73f16bc3" />

<img width="671" height="526" alt="Image" src="https://github.com/user-attachments/assets/a6c0e939-e957-4a8a-bcad-5628ae582aea" />

**Comment by SpartanJ** [comment]
I think the best option is configured from within the application and not make it dependant from the installer. It should be something very similar to what @NullPlane just shared.

I know this is useful but I have zero motivation to implement it because I don't use Windows. But it shouldn't be that hard to add. It would be cool to have more collaborators hehe.

**Comment by NullPlane** [comment]
> I think the best option is configured from within the application and not make it dependant from the installer. It should be something very similar to what [@NullPlane](https://github.com/NullPlane) just shared.
> 
> I know this is useful but I have zero motivation to implement it because I don't use Windows. But it shouldn't be that hard to add. It would be cool to have more collaborators hehe.

Add \[needs-collaboration\] flag.

I somehow anticipated you'd like that. But I actually think that this is very feasible and useful on Linux too. There are many ways to do it there, many better than Windows, maybe even some XDG-friendly, others may be handled through definition lists. And as ecode&mdash;which is rare but actually very cool&mdash;is a true portable app on linux too, until now, there is no way of integrating it other than manually or through third-party apps. 

Desktop icon, entries to shell paths (if the system supports automation of that), etc. being set up automatically is potentially very useful.
<details><summary><code>unnecessary rant</code></summary>
When I set ecode up on my uni's workplace, I had a few mischievous Arch users hitting at me why I set this manually and it doesn't ship automatically and I should just use VIM. Spoiler: they have to manually set up their entire OS hahaha</details>

**Comment by SpartanJ** [comment]
Implemented for all major platforms: Windows, macOS, Linux and FreeBSD.
